### PR TITLE
gate some libc usages under cfg(unix), drop os_info features

### PR DESCRIPTION
**Pull request by klensy**
Places few `libc` usages under `cfg(unix)`. That didn't remove it from tree, but still looks cleaner.
Drop features from os_info crate, as serde support currently unused.

### CODE REVIEW

File: credential/cargo-credential/Cargo.toml
Comment (weihanglo) [inline credential/cargo-credential/Cargo.toml:1]: You might need to bump a point version of `cargo-credential` to make CI green.

**Review by weihanglo — COMMENTED** [review]
This looks correct for usages in Cargo, though it doesn't really help shave off `libc` on Windows.

I am curious of what came to you for this, and also the benefit of it.

**Comment by klensy** [comment]
> This looks correct for usages in Cargo, though it doesn't really help shave off `libc` on Windows.
> 
> I am curious of what came to you for this, and also the benefit of it.

Initially i wanted to chop unused deps from cargo: found some target dependent `libc` usages and disabled them (but there a lot of libc around, so no complete remove), work was done and i decided to commit it anyway.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks.

@bors r+

**Pull request merged**